packages/docmanager-extension/src/index.ts

```diff
@@ -943,4 +954,33 @@ namespace Private {
     body.appendChild(lastCheckpointMessage);
     return body;
   }
+
+  /**
+   * Ask user for a checkpoint to revert to.
+   */
+  export async function getTargetCheckpoint(
+    checkpoints: Contents.ICheckpointModel[],
+    trans: TranslationBundle
+  ): Promise<Contents.ICheckpointModel | undefined> {
+    // the id could be too long to show so use the index instead
+    const indexSeparator = '.';
+    const items = checkpoints.map((checkpoint, index) => {
+      const isoDate = Time.format(checkpoint.last_modified);
+      const humanDate = Time.formatHuman(checkpoint.last_modified);
+      return `${index}${indexSeparator} ${isoDate} (${humanDate})`;
+    });
+
+    const selectedItem = (
+      await InputDialog.getItem({
+        items: items,
+        title: trans.__('Choose a checkpoint')
+      })
+    ).value;
+
+    if (!selectedItem) {
+      return;
+    }
+    const selectedIndex = parseInt(selectedItem.split(indexSeparator, 1)[0]);
```

Comment:
Maybe set `radix` to `10` in `parseInt`?

Reply:
Good point thank you.